Use custom iree.dev domain in links to documentation site.

ScottTodd opened the pull request:
Follow-up to https://github.com/openxla/iree/pull/15008 and some GitHub configuration changes that switched us to using a custom domain for our GitHub pages hosting.

Background:

* We acquired this domain for hosting https://perf.iree.dev/ years ago, with some aspirations to use the domain for other pages eventually.
* We tried to use the domain with our GitHub Pages content at one point, but ran into some issues that resulted in buggy redirects and certificate errors. A few weeks ago we tried that again and succeeded.

Note that we had also considered using the `docs` subdomain, i.e. `https://docs.iree.dev/`. That still seems tempting, but would require more domain configuration work. Until we use a subdomain like that or explore the workarounds in https://github.com/orgs/community/discussions/23184, the `www` part of URLs is required.

ScottTodd committed:
Use custom www.iree.dev domain in links to documentation site.

Mechanical replacement of `https://openxla.github.io/iree` with `https://www.iree.dev`. The `www` part is unfortunately required.

Background:

* We acquired this domain for hosting https://perf.iree.dev/ years ago, with some aspirations to use the domain for other pages eventually.
* We tried to use the domain with our GitHub Pages content at one point, but ran into some issues that resulted in buggy redirects and certificate errors. A few weeks ago we tried that again and succeeded.

Note that we had also considered using the `docs` subdomain, i.e. `https://docs.iree.dev/`. That still seems tempting, but would require more domain configuration work.

antiagainst reviewed (approved):
Nice. LGTM whenever you are ready. :)

ScottTodd commented on docs/website/mkdocs.yml:
```diff
@@ -1,5 +1,5 @@
 site_name: IREE
-site_url: https://openxla.github.io/iree/
+site_url: https://www.iree.dev/
```
Hmm... looks like the 404 page is broken with the current URL. Updating this site url may help.

https://www.iree.dev/404
```
 GET https://www.iree.dev/iree/assets/stylesheets/main.0e669242.min.css net::ERR_ABORTED 404 (Not Found)
 GET https://www.iree.dev/iree/assets/stylesheets/palette.85d0ee34.min.css net::ERR_ABORTED 404 (Not Found)
 GET https://www.iree.dev/iree/assets/stylesheets/extra.css net::ERR_ABORTED 404 (Not Found)
 GET https://www.iree.dev/iree/assets/stylesheets/openxla.css net::ERR_ABORTED 404 (Not Found)
 GET https://www.iree.dev/iree/assets/javascripts/bundle.78eede0e.min.js net::ERR_ABORTED 404 (Not Found)
 GET https://www.iree.dev/iree/assets/images/openxla-logo-lockup-white.svg 404 (Not Found)
```
Note the extra `iree/` in those paths. The URLs in the .html are of the form `<link rel="stylesheet" href="/iree/assets/stylesheets/extra.css">`

ScottTodd replied:
Nice, this fixed the 404 page (e.g. https://iree.dev/404)

ScottTodd committed:
Revert "Use custom www.iree.dev domain in links to documentation site."

This reverts commit 63155fc154c8f3111ae7fe71982921d041520f98.

ScottTodd committed:
Merge remote-tracking branch 'upstream/main' into docs-url

ScottTodd committed:
Use custom iree.dev domain in links to documentation site.

Mechanical replacement of `https://openxla.github.io/iree` with `https://iree.dev`.

Background:

* We acquired this domain for hosting https://perf.iree.dev/ years ago, with some aspirations to use the domain for other pages eventually.
* We tried to use the domain with our GitHub Pages content at one point, but ran into some issues that resulted in buggy redirects and certificate errors. A few weeks ago we tried that again and succeeded.

Note that we had also considered using the `docs` subdomain, i.e. `https://docs.iree.dev/`. That still seems tempting, but would require more domain configuration work.
